### PR TITLE
Remove 'Enterprise Features' page

### PR DESCRIPTION
**Pull request by rmloveland**
Addresses DOC-11692

### CODE REVIEW

**Comment by rmloveland** [comment]
@jaiayu adding you to review as an FYI

most removals of links to this page happened in https://github.com/cockroachdb/docs/pull/19001

this removes the actual 'Enterprise Features' page itself, along with the rest of the links to it 

(with one exception, a link that is being removed in https://github.com/cockroachdb/docs/pull/19002) 

this will also be immediately backported to v23.1, v23.2, v24.1, v24.2 once we're happy with it

**Comment by rmloveland** [comment]
@taroface adding you concurrently for speed - i don't think (?) there is much to review here

**Review by mdlinville — APPROVED** [review]

File: src/current/v24.3/choose-a-deployment-option.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ Cockroach Labs offers several options for clusters: CockroachDB {{ site.data.pro
         <li><b>Resource isolation</b>: Shared CockroachDB software and infrastructure. Data is protected and not shared between deployments.</li>
         <li><b><a href="../cockroachcloud/cloud-org-sso.html">Single Sign-On (SSO)</a></b>: Authentication enforcement using a centralized identity managed by an IdP for clusters and organizations.</li>
         <li><b>Support</b>: Provided by CockroachDB <a href="https://forum.cockroachlabs.com/">community forum</a> and public <a href="https://cockroachdb.slack.com/">Slack workspace</a>.</li>
-        <li><b><a href="enterprise-licensing.html">Enterprise Features</a></b>: Enabled</li>
+        <li><b>Enterprise Features</b>: Enabled</li>
```
Comment (taroface) [inline src/current/v24.3/choose-a-deployment-option.md:34]:
```suggestion
        <li><b>Enterprise Features</b>: Enabled.</li>
```

**Review by taroface — APPROVED** [review]
LGTM, 1 nit

File: src/current/v24.3/choose-a-deployment-option.md (hunk 1 above)
Comment (rmloveland) [reply src/current/v24.3/choose-a-deployment-option.md:34]: thanks, ✅

**Comment by rmloveland** [comment]
Gonna fix the busted link here to get this in ahead of #19002 

will then backport this ASAP

File: src/current/v24.3/changefeed-examples.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ toc: true
 docs_area: stream_data
 ---
 
-This page provides step-by-step examples for using Core and {{ site.data.products.enterprise }} changefeeds. Creating {{ site.data.products.enterprise }} changefeeds is available on CockroachDB {{ site.data.products.basic }}, {{ site.data.products.standard }}, {{ site.data.products.advanced }}, and with an [{{ site.data.products.enterprise }} license](enterprise-licensing.html) on CockroachDB {{ site.data.products.core }} clusters. Basic changefeeds are available in all products.
+This page provides step-by-step examples for using Core and {{ site.data.products.enterprise }} changefeeds. Creating {{ site.data.products.enterprise }} changefeeds is available on CockroachDB {{ site.data.products.basic }}, {{ site.data.products.standard }}, {{ site.data.products.advanced }}, and with an [{{ site.data.products.enterprise }} license](licensing-faqs.html#types-of-licenses) on CockroachDB {{ site.data.products.core }} clusters. Basic changefeeds are available in all products.
```
Comment (mdlinville) [inline src/current/v24.3/changefeed-examples.md:8]: Nit: Liquid-style link?

**Review by mdlinville — APPROVED** [review]

File: src/current/v24.3/changefeed-examples.md (hunk 2 above)
Comment (rmloveland) [reply src/current/v24.3/changefeed-examples.md:8]: it's too late the merge train is rolling!!!!!! sorry

**Pull request merged**